tiny_dnn/core/framework/tensor.h

```diff
@@ -335,7 +318,36 @@ class Tensor {
                                                           shape_.end()));
     }
 
+    Tensor view(std::initializer_list<size_t> const &start,
+                std::initializer_list<size_t> const &shape) {
+	if (start.size() > kDimensions || shape.size() > kDimensions) {
+            throw nn_error("Overpassed number of existing dimensions.");
+	}
+        return Tensor(storage_ptr_, compute_offset(start, shape_), shape);
+    }
+
+    bool isView() const {
+        return size_ != storage_ptr_->size();
```

Comment:
@Randl I have been checking and not all views have `offset != 0` since we can take a view starting from zero position